### PR TITLE
T-52: Moving entrypoint to godot

### PR DESCRIPTION
**Pull request by MasterHamster42**

### CODE REVIEW

File: core/src/communication/application/schema.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    class Config:
+        allow_mutation = False
+
+    class Resources(BaseModel):
+        class Config:
+            allow_mutation = False
+
+        number_of_workers: PositiveInteger
+        number_of_env_per_worker: PositiveInteger
+        training_checkpoint_frequency: PositiveInteger
```
Comment (MartynaSokolowska) [inline core/src/communication/application/schema.py:37]: Tu można by dodać liczbe zapisywanych wstecz, żeby mieć jakąś historie, chyba że zakładamy że wszystkie zapisujemy (?)

File: core/src/communication/application/schema.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    class TrainingVariables(BaseModel):
+        class Config:
+            allow_mutation = False
+
+        training_iterations: PositiveInteger
+        training_batch_size: PositiveInteger
+        learning_rate: PositiveFiniteFloat
+        entropy_coefficient: PositiveFiniteFloat
```
Comment (MartynaSokolowska) [inline core/src/communication/application/schema.py:46]: I tu chcemy tylko to lr i entorpy_coeff, przekazywane rozumiem z Godota a reszta bedzie ustawiona, żeby pasowała do naszego środowiska? Jeszcze chyba potrzebujemy wyboru modelu do trenowania?

File: core/src/utils/training_handler.py [hunk 3]
```diff
@@ -29,10 +29,10 @@ def __init__(
         self.model_name = model_name
         self.environment_cls = environment_cls
 
-    def train(self):
-        training_settings = get_settings().training
+    def train(self, training_settings: TrainingSettings):
         save_interval = training_settings.training_checkpoint_frequency
-        model_dir = get_path()
+        # model_dir = get_path() # infinitive loop :(
```
Comment (MartynaSokolowska) [inline core/src/utils/training_handler.py]: :(

File: core/src/app.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            raw_settings = handler.receive()  # hmm?
+            core_settings = CoreSettingsSchema.model_validate_json(raw_settings)
+            handler.send(bytes(1))
+            self.registry.put(CoreSettingsSchema, core_settings)
+
+            # For now just training?
+            self.train()
+
+    @classmethod
+    def startup_ray(cls):
```
Comment (MartynaSokolowska) [inline core/src/app.py:61]: Można by to wyrzucić stąd i zrobić klase osobną do ustawień raya (?)

Comment (MartynaSokolowska) [reply core/src/app.py:61]: i też pipe handler i godot app handler, co by było bradziej czytelne, nie wiem na ile jest to tu potrzebne

File: core/src/app.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        ray.shutdown()
+        ray.init(runtime_env={"worker_process_setup_hook": configure_logging}, include_dashboard=True, num_gpus=1)
```
Comment (MartynaSokolowska) [inline core/src/app.py:63]: tu mi często wywala, że za mało pamięci zabiera ta inicjalizacja raya to mozna by dać try-except i retry

File: core/src/communication/application/schema.py (hunk 2 above)
Comment (MartynaSokolowska) [reply core/src/communication/application/schema.py:46]: I też to Custom model zrobiłam, to je łatwo przerobić że można wybrać ilość i wielkości warstw to by też można było wybierać w godocie i tez dropout i inne tego typu rzeczy.

**Review by Kiszkacy — APPROVED** [review]
